File: playwright-java/src/test/java/testtracks/pages/restfulbooker/admin/AdminLoginPage.java
package testtracks.pages.restfulbooker.admin;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;
import testtracks.configuration.TestSettings;
import testtracks.ui.common.BasePage;

public final class AdminLoginPage extends BasePage {
  private static final String ADMIN_ROUTE = "admin";
  private static final String USERNAME_LABEL = "Username";
  private static final String PASSWORD_LABEL = "Password";
  private static final String LOGIN_BUTTON_NAME = "Login";

  private Locator username() {
    return page.getByLabel(USERNAME_LABEL);
  }

  private Locator password() {
    return page.getByLabel(PASSWORD_LABEL);
  }

  private Locator loginButton() {
    return page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName(LOGIN_BUTTON_NAME));
  }

  public AdminLoginPage(Page page, TestSettings settings) {
    super(page, settings);
  }

  public void logIn(String username, String password) {
    open(ADMIN_ROUTE);
    username().fill(username);
    password().fill(password);
    loginButton().click();
  }
}

File: playwright-csharp/tests/Pages/RestfulBooker/Admin/AdminLoginPage.cs
using System.Text.RegularExpressions;
using Microsoft.Playwright;
using TestTracks.Playwright.CSharp.Configuration;
using TestTracks.Playwright.CSharp.UI.Common;

namespace TestTracks.Playwright.CSharp.Specs.Pages.RestfulBooker.Admin;

public sealed class AdminLoginPage : BasePage
{
    private const string AdminRoute = "admin";
    private const string UsernameLabel = "Username";
    private const string PasswordLabel = "Password";
    private const string LoginButtonName = "Login";

    private ILocator Username => Page.GetByLabel(new Regex(UsernameLabel, RegexOptions.IgnoreCase));

    private ILocator Password => Page.GetByLabel(new Regex(PasswordLabel, RegexOptions.IgnoreCase));

    private ILocator LoginButton =>
        Page.GetByRole(
            AriaRole.Button,
            new() { NameRegex = new Regex(LoginButtonName, RegexOptions.IgnoreCase) });

    public AdminLoginPage(IPage page, TestSettings settings)
        : base(page, settings)
    {
    }

    public async Task LogIn(string username, string password)
    {
        await OpenAsync(AdminRoute);
        await Username.FillAsync(username);
        await Password.FillAsync(password);
        await LoginButton.ClickAsync();
    }
}
